airbyte-cdk/python/airbyte_cdk/sources/declarative/parsers/model_to_component_factory.py

```diff
@@ -791,14 +791,17 @@ def create_simple_retriever(
             else NoPagination(parameters={})
         )
 
+        stream_slicer = stream_slicer or SinglePartitionRouter(parameters={})
+        cursor = stream_slicer if isinstance(stream_slicer, Cursor) else None
```

Comment:
This block feels a bit weird. In the case of a `DatetimeBasedCursor`, this results in both the `stream_slicer` and the `cursor` getting set as the same component. So I like the idea of the Cursor and the StreamSlicer having distinct responsibilities, but when the Cursor is a subclass of the StreamSlicer, it does lead to this duplicative assignment. So it got me thinking about our object model a bit.

Its a little late so this could just be very stream of conscious, but I think it could be an interesting thought to even further decouple the concepts of the date ranges and the cursor state. So instead of Cursor inheriting StreamSlicer, they would be their own distinct interfaces. For example, we would have the date cursor only used to maintain state and a SinglePartitionRouter where the slice is just over the entire state -> present. And the SimpleRetriever is truly composed of one StreamSlicer and one Cursor. 

This would lead to a bigger refactor in the SimpleRetriever and DatetimeBasedCursor since they'd have to work across interface boundaries to persist state when we finish a range. Maybe we could discuss this synchronously if this is an idea worth fleshing out

Reply:
Discussed this w/ maxime offline. we think a greater refactor to truly decouple the Cursor inheritance from the StreamSlicer might be a good idea later down the line, but it is at this moment not strictly necessary to solve in the immediate for what is being implemented for per-stream

The conceptual change of decoupling cursors and stream slicers is that we can think of them as distinctive pieces of functionality in that a cursor is used to support state/incremental syncs and stream slicing handles creating date ranges or other slices. In theory you could have incremental syncs w/ one giant slice or date range slices over the entire time period. The simple retriever would then act as an orchestrator of components like it already does. If we do find a use case for this then we can tie the refactor to that need.